nomad/structs/structs.go

```diff
@@ -5731,7 +5731,8 @@ func (a *Allocation) RescheduleEligible(reschedulePolicy *ReschedulePolicy, fail
 }
 
 // LastEventTime is the time of the last task event in the allocation.
-// It is used to determine allocation failure time.
+// It is used to determine allocation failure time. If the FinishedAt field
```

Comment:
Just to capture the final decision. We will rely on the following time values in order, only skipping if not set: FinishedAt, last task event timestamp, alloc modify time